pkgs/development/rocm-modules/6/tensile/default.nix

```diff
@@ -34,6 +35,19 @@ buildPythonPackage rec {
     joblib
   ];
 
+  patches = [
+    (fetchpatch {
+      name = "Extend-Tensile-HIP-ISA-compatibility.patch";
+      url = "https://github.com/GZGavinZhao/Tensile/commit/855cb15839849addb0816a6dde45772034a3e41f.patch";
+      hash = "sha256-d+fVf/vz+sxGqJ96vuxe0jRMgbC5K6j5FQ5SJ1e3Sl8=";
+    })
+    (fetchpatch {
+      name = "Don-t-copy-file-twice-in-copyStaticFiles.patch";
+      url = "https://github.com/GZGavinZhao/Tensile/commit/9e14d5a00a096bddac605910a0e4dfb4c35bb0d5.patch";
```

Comment:
In the original rocBLAS recipe, Tensile was rewrapped because `# Tensile REALLY wants to write to the nix directory if we include it normally`. More precisely, during setup phase, Tensile copies the C++ headers it provides into the build directory of rocBLAS. Then, during build phase, Tensile copies it again. _However_, for some reason, despite that it is copying the same file, copying would fail with permission error during build phase but not setup phase. I'm very confused by why this happens, but I've logged the copy commands and verified that it is indeed copying the same file. This patch basically let Tensile only copy the headers if source and destination have different contents. So yes, this check is specifically for nixpkgs to allow us not needing to rewrap Tensile when building `rocblas`.

Reply:
I'd also note that I can't build rocBLAS on the `master` branch because rocBLAS cannot find Tensile during setup phase, so that's another reason to use this patch and not wrap Tensile anymore in rocBLAS.

Reply:
This makes sense.

I think prevents source from being writable in later build phases on purpose.